xcdat/spatial.py

```diff
@@ -441,12 +408,34 @@ def _get_longitude_weights(
         -------
         xr.DataArray
             The longitude axis weights.
+
+        Raises
+        ------
+        ValueError
+            If the there are multiple instances in which the
+            domain_bounds[:, 0] > domain_bounds[:, 1]
         """
         p_meridian_index: Optional[np.ndarray] = None
         d_bounds = domain_bounds.copy()
 
+        # check if there is more than one potential prime meridian cell
+        # there should only be one for rectilinear data
+        pmcells = np.where(domain_bounds[:, 1] - domain_bounds[:, 0] < 0)[0]
+        if len(pmcells) > 1:
+            raise ValueError(
+                "More than one longitude bound is out of order. Only one bound \n\
+                              spanning the prime meridian is permitted"
+            )
+        # convert longitude bounds to 360 degree domain
+        d_bounds: xr.DataArray = self._swap_lon_axis(d_bounds, to=360)  # type: ignore
+        # check for bounds spanning prime meridian
+        p_meridian_index = _get_prime_meridian_index(d_bounds)
+        # if bounds span a prime meridian, ensure bounds are organized to span 0-360
+        # (without losing weight)
+        if p_meridian_index is not None:
+            d_bounds = _align_lon_bounds_to_360(d_bounds, p_meridian_index)
```

Comment:
```suggestion
        # The longitude bounds are converted to the 360 degree domain if it
        # is not already. Afterwards, the index of the bounds with a prime
        # meridian cell is extracted (if it exists). If a prime meridian cell
        # exists, bounds are organized to span 0-360 in ascending order (without
        # losing weight).
        d_bounds: xr.DataArray = self._swap_lon_axis(d_bounds, to=360)  # type: ignore
        p_meridian_index = _get_prime_meridian_index(d_bounds)
        if p_meridian_index is not None:
            d_bounds = _align_lon_bounds_to_360(d_bounds, p_meridian_index)
```

Reply:
Joined comments together and updated to specify "bounds are organized to span 0-360 in ascending order".